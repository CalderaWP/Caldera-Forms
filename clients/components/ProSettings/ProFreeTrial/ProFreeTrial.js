import React from 'react';
import propTypes from 'prop-types';
import classNames from 'classnames';
import {Button} from 'react';

/**
 * Create the ProFreeTrial UI
 * @param {Object} props
 * @return {*}
 * @constructor
 */
export const ProFreeTrial = (props) => {
	const documentationHref = `https://calderaforms.com/doc/caldera-forms-pro-getting-started/?utm_source=wp-admin&utm_campaign=pro-screen&utm_term=not-connected`;
	const trialHref = `https://calderaforms.com/checkout?edd_action=add_to_cart&download_id=64101&edd_options[price_id]=1?utm_source=wp-admin&utm_campaign=pro-screen&utm_term=not-connected`;
	return(
		<div
			className={classNames(props.className,ProFreeTrial.classNames.wrapper)}
		>
			<p>Ready to try Caldera Forms Pro? Plans start at just 14.99/ month with a 7 day free trial.</p>
			<div>
				<a
					href={documentationHref}
					target={'_blank'}

					className={'button'}
				>
					Documentation
				</a>
				<a
					target={'_blank'}
					href={trialHref}
					className={'button button-primary'}
				>
					Start Free Trial
				</a>
			</div>
		</div>
	)
};

/**
 * Prop types for the ProFreeTrial component
 * @type {{}}
 */
ProFreeTrial.propTypes = {
	classNames: propTypes.string
};


/**
 * Class names used in ProFreeTrial component
 * @type {{wrapper: string}}
 */
ProFreeTrial.classNames = {
	wrapper: 'caldera-forms-pro-free-trial'
}
