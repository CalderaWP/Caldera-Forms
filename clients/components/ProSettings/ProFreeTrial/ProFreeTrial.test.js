import renderer from 'react-test-renderer';
import React from 'react';
import { shallow } from 'enzyme/build';
import Enzyme from 'enzyme/build';
import Adapter from 'enzyme-adapter-react-16/build';
import {ProFreeTrial} from "./ProFreeTrial";

Enzyme.configure({ adapter: new Adapter() });

describe( 'ProFreeTrial component', () => {
	it( 'Matches snapshot with minimal props', () => {
		expect(
			renderer.create(
				<ProFreeTrial/>
			).toJSON()
		).toMatchSnapshot()
	});

	it( 'Is wrapped in the right class', () => {
		expect(
			shallow(
				<ProFreeTrial/>
			).find( '.' + ProFreeTrial.classNames.wrapper )
				.length
		).toEqual(1)
	});
});