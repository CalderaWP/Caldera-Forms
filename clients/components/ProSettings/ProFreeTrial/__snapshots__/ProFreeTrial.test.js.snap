// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProFreeTrial component Matches snapshot with minimal props 1`] = `
<div
  className="caldera-forms-pro-free-trial"
>
  <p>
    Ready to try Caldera Forms Pro? Plans start at just 14.99/ month with a 7 day free trial.
  </p>
  <div>
    <a
      className="button"
      href="https://calderaforms.com/doc/caldera-forms-pro-getting-started/?utm_source=wp-admin&utm_campaign=pro-screen&utm_term=not-connected"
      target="_blank"
    >
      Documentation
    </a>
    <a
      className="button button-primary"
      href="https://calderaforms.com/checkout?edd_action=add_to_cart&download_id=64101&edd_options[price_id]=1?utm_source=wp-admin&utm_campaign=pro-screen&utm_term=not-connected"
      target="_blank"
    >
      Start Free Trial
    </a>
  </div>
</div>
`;
